addons/web/static/src/js/widgets/model_field_selector.js

```diff
@@ -0,0 +1,314 @@
+                    e.preventDefault();
+                    var $active = this.$("li.o_field_selector_item.active");
+                    var $to = $active[e.which === $.ui.keyCode.DOWN ? "next" : "prev"](".o_field_selector_item");
+                    if ($to.length) {
+                        $active.removeClass("active");
+                        $to.addClass("active");
+                        this.$popover.focus();
+
+                        var $page = $to.closest(".o_field_selector_page");
+                        var full_height = $page.height();
+                        var el_position = $to.position().top;
+                        var el_height = $to.outerHeight();
+                        var current_scroll = $page.scrollTop();
+                        if (el_position < 0) {
+                            $page.scrollTop(current_scroll - el_height);
+                        } else if (full_height < el_position + el_height) {
+                            $page.scrollTop(current_scroll + el_height);
+                        }
+                    }
+                    break;
+                case $.ui.keyCode.RIGHT:
+                    if (inputHasFocus) break;
+                    e.preventDefault();
+                    var name = this.$("li.o_field_selector_item.active").data("name");
+                    if (name) {
+                        var field = this._getLastPageField(name);
+                        if (field.relation) {
+                            this.goToNextPage(field);
+                        }
+                    }
+                    break;
+                case $.ui.keyCode.LEFT:
+                    if (inputHasFocus) break;
+                    e.preventDefault();
+                    this.goToPrevPage();
+                    break;
+                case $.ui.keyCode.ESCAPE:
+                    e.stopPropagation();
+                    this.hidePopover();
+                    break;
+                case $.ui.keyCode.ENTER:
+                    if (inputHasFocus) break;
+                    e.preventDefault();
+                    this.selectField(this._getLastPageField(this.$("li.o_field_selector_item.active").data("name")));
+                    break;
+            }
+        },
+    },
+    /// The ModelFieldSelector requires a model and a initial field chain to work with.
+    /// @param model - a string with the model name (e.g. "res.partner")
+    /// @param chain - a string with the initial field chain (e.g. "company_id.name")
+    /// @param filters - an object which contains a series of filters to use
+    init: function (parent, model, chain, filters) {
+        this._super.apply(this, arguments);
+
+        this.model = model;
+        this.chain = chain;
+        this.options = {filters: filters || {}};
+
+        this.pages = [];
+        this.selectedField = false;
+        this.isSelected = true;
+        this.dirty = false;
+
+        this.debug = session.debug;
```

Comment:
as much as possible, widgets should be instantiated with all the necessary information, not fetch the information themselves.